core/libp2p/stream/stream.hpp

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_STREAM_HPP
+#define KAGOME_STREAM_HPP
+
+#include <rxcpp/rx-observable.hpp>
+#include "common/result.hpp"
+#include "libp2p/basic_interfaces/writable.hpp"
+#include "libp2p/common/network_message.hpp"
+
+namespace libp2p::stream {
+  /**
+   * Stream between two peers in the network
+   */
+  class Stream : public basic::Writable {
+    /**
+     * Read messages from the stream
+     * @return observable to messages, received by that stream
+     */
+    virtual rxcpp::observable<common::NetworkMessage> read() const = 0;
```

Comment:
Why there's a `Writable`, but no `Readable`?

Reply:
Because signatures of `read`-s in two classes using it are different:
- `boost::optional<common::NetworkMessage>  Connection::read()`
- `rxcpp::observable<common::NetworkMessage> Stream::read()`
I assume that a `Connection`'s one is a one-time operation, returning something and finishing, while the `Stream`'s a stream of messages.